test/functional/testlib.bash

```diff
@@ -1888,23 +1891,33 @@ create_bundle() { # swupd_function
 				bundle_dir=$(create_dir "$files_path")
 				add_to_manifest -p "$manifest" "$bundle_dir" "$fdir"
 			fi
+		else
+			fdir=""
 		fi
-		bundle_link=$(create_link "$files_path")
-		sudo tar -C "$files_path" -rf "$version_path"/pack-"$bundle_name"-from-0.tar --transform "s,^,staged/," "$(basename "$bundle_link")"
-		add_to_manifest "$manifest" "$bundle_link" "$val"
-		# Add the file pointed by the link to the zero pack of the bundle
-		pfile=$(basename "$(readlink -f "$bundle_link")")
+		# first create the file that will be pointed by the symlink
+		pfile=$(create_file "$files_path")
+		pfile_name=$(generate_random_name file_)
+		pfile_path="$fdir"/"$pfile_name"
+		# temporarily rename the file to the name it will have in the file system
+		# so we can create the symlink with the correct name
+		sudo mv "$pfile" "$(dirname "$pfile")"/"$pfile_name"
```

Comment:
Perhaps `"$(dirname "$pfile")"/"$pfile_name"` can be stored in a local variable, and then referenced three times? I think this would make these three lines easier to read.